src/components/lyne-autocomplete/lyne-autocomplete.tsx

```diff
@@ -0,0 +1,236 @@
+  },
+  tag: 'lyne-autocomplete'
+})
+
+export class LyneAutocomplete {
+
+  /**
+   * Items to show in the autocomplete interface. You should pass a stringified
+   * array of objects, containing the `text` key for each object with an
+   * appropriate value.
+   */
+  @Prop() public items!: string;
+
+  /**
+   * The value to use as default value for the input. The input value or the
+   * selected autocomplete value is reflected to this attribute.
+   */
+  @Prop({
+    reflect: true
+  }) public value?: string;
+
+  /**
+   * Id which is sent as the id in the eventDetail payload when a
+   * value is selected
+   */
+  @Prop() public eventId?: string;
+
+  /**
+   * Autocomplete id. If you use multiple instances on a page,
+   * you should set individual id's to each of them.
+   */
+  @Prop() public autocompleteId? = 'autocomplete-list';
+
+  /**
+   * Define if borderless variant of autocomplete input should be used. See
+   * documentation of lyne-text-input for details.
+   */
+  @Prop() public inputBorderless?: boolean;
+
+  /**
+   * Debounce timeout to use for the input. See documentation of
+   * lyne-text-input for details.
+   */
+  @Prop() public inputDebounceTimeout? = 200;
+
+  /**
+   * Name attribute for the input element. See lyne-text-input for details.
+   */
+  @Prop() public inputName!: string;
+
+  /**
+   * Label attribute for the input element. See lyne-text-input for details.
+   */
+  @Prop() public inputLabel!: string;
+
+  /**
+   * Placeholder attribute for the input element. See lyne-text-input for
+   * details.
+   */
+  @Prop() public inputPlaceholder?: string;
+
+  /**
+   * Determine if the input label should be visible.
+   * See lyne-text-input for
+   * details.
+   */
+  @Prop() public inputLabelVisible?: boolean;
+
+  @State() private _inputValue: string;
+  @State() private _isVisible = false;
+
+  @Element() private _element: HTMLElement;
+
+  private _inputElement!: HTMLLyneTextInputElement;
+  private _list!: HTMLLyneAutocompleteListElement;
+
+  private _handleKeyPress = (evt): void => {
+    const {
+      key
+    } = evt;
+
+    const event = new CustomEvent(events.keypress, {
+      bubbles: false,
+      composed: false,
+      detail: key
+    });
+
+    /**
+     * lyne-autocomplete-list listens to this event
+     */
+    this._list.dispatchEvent(event);
+  };
+
+  private _selectInputText = (): void => {
+    const event = new CustomEvent('select', {
+      bubbles: false,
+      composed: false
+    });
+
+    /**
+     * lyne-text-input listens to this event
+     */
+    this._inputElement.dispatchEvent(event);
+  };
+
+  private _focusInputElement = (): void => {
+    const event = new CustomEvent('focus', {
+      bubbles: false,
+      composed: false
+    });
+
+    /**
+     * lyne-text-input listens to this event
+     */
+    this._inputElement.dispatchEvent(event);
+  };
+
+  private _handleFocus = (): void => {
+    this._isVisible = true;
```

Comment:
I thought about leaving this to the consumer. He/she is responsible to provide the items to show as the autocomplete. As soon as we start making a variant for Timetable Input Form, we need to extend the item objects passed by the consumer, where he/she will mark an item as favorite, one from the history, or one currently coming from the api. otherwise i'd be afraid that we would build too much business logic into this component.

Reply:
Made some changes to this: `items` is now optional. The autocomplete list is only shown if the consumer has provided valid `items`. I would leave the rest of the logic up to the consumer.